### PR TITLE
Make the sync segment request idempotent in the data node

### PR DESCRIPTION
**Pull request by SimFG**
issue: #20706
Signed-off-by: SimFG <bang.fu@zilliz.com>

### CODE REVIEW

File: internal/datanode/data_node.go [hunk 1]
```diff
@@ -944,7 +943,7 @@ func (node *DataNode) SyncSegments(ctx context.Context, req *datapb.SyncSegments
 		numRows:      req.GetNumOfRows(),
 	}
 
-	err = channel.InitPKstats(ctx, targetSeg, req.GetStatsLogs(), tsoutil.GetCurrentTime())
+	err := channel.InitPKstats(ctx, targetSeg, req.GetStatsLogs(), tsoutil.GetCurrentTime())
```
Comment (soothing-rain) [inline internal/datanode/data_node.go:946]: If we have invalid segment in compactFrom, PK stats are still generated here?

Comment (SimFG) [reply internal/datanode/data_node.go:946]: The information related to compactFrom is not involved. The bloomFilterFile is mainly read according to the path about the compactTo, so repeating init should be no problem.

**Pull request merged**